python/packages/autogen-ext/src/autogen_ext/tools/mcp/_session.py

```diff
@@ -26,3 +31,105 @@ async def create_mcp_server_session(
         async with sse_client(**server_params.model_dump()) as (read, write):
             async with ClientSession(read_stream=read, write_stream=write) as session:
                 yield session
+
+
+class McpSessionConfig(BaseModel):
+    server_params: McpServerParams
+
+
+class McpSessionActor(ComponentBase[BaseModel], Component[McpSessionConfig]):
+    component_type = "mcp_session_actor"
+    component_config_schema = McpSessionConfig
+    component_provider_override = "autogen_ext.tools.mcp.McpSessionActor"
+
+    server_params: McpServerParams
+    _actor: Any = PrivateAttr(default=None)
+    # actor: Any = PrivateAttr(default=None)
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    def __init__(self, server_params: McpServerParams) -> None:
+        self.server_params: McpServerParams = server_params
+        self.name = "mcp_session_actor"
+        self.description = "MCP session actor"
+        self._command_queue: asyncio.Queue[Dict[str, Any]] = asyncio.Queue()
+        self._actor_task: asyncio.Task[Any] | None = None
+        self._shutdown_future: asyncio.Future[Any] | None = None
+        self._active = False
+        atexit.register(self._sync_shutdown)
+
+    async def initialize(self) -> None:
+        if not self._active:
+            self._active = True
+            self._actor_task = asyncio.create_task(self._run_actor())
+
+    async def call(self, name: str, kwargs: Dict[str, Any]) -> Any:
+        if not self._active:
+            raise RuntimeError("MCP Actor not running, call initialize() first")
+        if self._actor_task and self._actor_task.done():
+            raise RuntimeError("MCP actor task crashed", self._actor_task.exception())
+        fut: asyncio.Future[Any] = asyncio.Future()
+        await self._command_queue.put({"type": "call", "name": name, "args": kwargs, "future": fut})
+        res = await fut
+        return res
+
+    async def close(self) -> None:
+        if not self._active or self._actor_task is None:
+            return
+        self._shutdown_future = asyncio.Future()
+        await self._command_queue.put({"type": "shutdown", "future": self._shutdown_future})
+        await self._shutdown_future
+        await self._actor_task
+        self._active = False
+
+    async def _run_actor(self) -> None:
+        try:
+            async with create_mcp_server_session(self.server_params) as session:
+                await session.initialize()
+                while True:
+                    cmd = await self._command_queue.get()
```

Comment:
Do we need a queue for processing the session commands?

Reply:
Same issue from under that.
I will summary it at the under comment.